src/App/Document.cpp

```diff
@@ -1261,6 +1261,52 @@ bool Document::isTransactionEmpty() const
     return true;
 }
 
+void Document::emptyDocument()
+{
+    this->d->activeObject = 0;
+
+    if(this->d->objectArray.size()) {
+        GetApplication().signalDeleteDocument(*this);
+        this->d->objectArray.clear();
+        for(auto &v : this->d->objectMap) {
+            v.second->setStatus(ObjectStatus::Destroy, true);
+            delete(v.second);
+        }
+        this->d->objectMap.clear();
+        this->d->objectIdMap.clear();
+        GetApplication().signalNewDocument(*this,false);
+    }
+
+    Base::FlagToggler<> flag(_IsRestoring,false);
+
+    setStatus(Document::PartialDoc,false);
+
+    this->d->clearRecomputeLog();
+    this->d->objectArray.clear();
+    this->d->objectMap.clear();
+    this->d->objectIdMap.clear();
+    this->d->lastObjectId = 0;
+}
+
+void Document::resetTouched()
+{
+/*

```

Comment:
Can we do something about this commented out section?

Reply:
You are right, that is dead code coming from the fact that between the time I wrote the code and pushed for the PR, there has been updates into document management. I removed that code and calls